### ISSUE TITLE
Testing Machine.wait_boot() races with shutdown

### ISSUE THREAD

**Issue by stefwalter** [cockpit-project/cockpit#2628]
Since wait_boot() now waits for IP to be accessible, we cannot simply use it when waiting for a reboot in the test. It races with the shutdown.

**Comment by stefwalter** [comment]
```
Wrote check-shutdown-restart-testBasic-FAIL.png
ssh_exchange_identification: read: Connection reset by peer
Essh_exchange_identification: read: Connection reset by peer
E
======================================================================
ERROR: testBasic (__main__.TestShutdownRestart)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "./check-shutdown-restart", line 40, in testBasic
    self.start_cockpit()
  File "/home/hubbot/hubbot/602256341cf5b931714c21be14d4b7cfc1897b29_r7_x86-64/test/testlib.py", line 478, in start_cockpit
    self.machine.execute("systemctl start cockpit-testing.socket")
  File "/home/hubbot/hubbot/602256341cf5b931714c21be14d4b7cfc1897b29_r7_x86-64/test/testvm.py", line 335, in execute
    raise subprocess.CalledProcessError(proc.returncode, command)
CalledProcessError: Command 'systemctl start cockpit-testing.socket' returned non-zero exit status 255

======================================================================
ERROR: testBasic (__main__.TestShutdownRestart)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/hubbot/hubbot/602256341cf5b931714c21be14d4b7cfc1897b29_r7_x86-64/test/testlib.py", line 431, in tearDown
    self.copy_journal("FAIL")
  File "/home/hubbot/hubbot/602256341cf5b931714c21be14d4b7cfc1897b29_r7_x86-64/test/testlib.py", line 573, in copy_journal
    m.download_dir("/var/log/journal", dir)
  File "/home/hubbot/hubbot/602256341cf5b931714c21be14d4b7cfc1897b29_r7_x86-64/test/testvm.py", line 400, in download_dir
    subprocess.check_call(cmd)
  File "/usr/lib64/python2.7/subprocess.py", line 542, in check_call
    raise CalledProcessError(retcode, cmd)
CalledProcessError: Command '['scp', '-i', '/home/hubbot/hubbot/602256341cf5b931714c21be14d4b7cfc1897b29_r7_x86-64/test/identity', '-o', 'StrictHostKeyChecking=no', '-o', 'UserKnownHostsFile=/dev/null', '-r', 'root@10.111.111.14:/var/log/journal', 'check-shutdown-restart-testBasic-10.111.111.14-FAIL.journal']' returned non-zero exit status 1

----------------------------------------------------------------------
Ran 1 test in 25.083s
```

**Comment by stefwalter** [comment]
This is a hard one to fix. The regression was introduced once we stopped using the terminal for noting when a machine boots.

**Comment by stefwalter** [comment]
I think this one is also related:

```
OK
Wrote check-shutdown-restart-testBasic-FAIL.png
Write failed: Broken pipe
ssh: connect to host 10.111.111.14 port 22: Connection timed out
ssh: connect to host 10.111.111.14 port 22: No route to host
ssh: connect to host 10.111.111.14 port 22: No route to host
ssh: connect to host 10.111.111.14 port 22: No route to host
ssh: connect to host 10.111.111.14 port 22: No route to host
ssh: connect to host 10.111.111.14 port 22: No route to host
ssh: connect to host 10.111.111.14 port 22: No route to host
ssh: connect to host 10.111.111.14 port 22: No route to host
ssh: connect to host 10.111.111.14 port 22: No route to host
ssh: connect to host 10.111.111.14 port 22: No route to host
ssh: connect to host 10.111.111.14 port 22: No route to host
ssh: connect to host 10.111.111.14 port 22: No route to host
ssh: connect to host 10.111.111.14 port 22: No route to host
ssh: connect to host 10.111.111.14 port 22: No route to host
ssh: connect to host 10.111.111.14 port 22: No route to host
ssh: connect to host 10.111.111.14 port 22: No route to host
ssh: connect to host 10.111.111.14 port 22: No route to host
ssh: connect to host 10.111.111.14 port 22: No route to host
ssh: connect to host 10.111.111.14 port 22: No route to host
ssh: connect to host 10.111.111.14 port 22: No route to host
ssh: connect to host 10.111.111.14 port 22: No route to host
ssh: connect to host 10.111.111.14 port 22: No route to host
ssh: connect to host 10.111.111.14 port 22: No route to host
ssh: connect to host 10.111.111.14 port 22: No route to host
ssh: connect to host 10.111.111.14 port 22: No route to host
ssh: connect to host 10.111.111.14 port 22: No route to host
ssh: connect to host 10.111.111.14 port 22: No route to host
ssh: connect to host 10.111.111.14 port 22: No route to host
ssh: connect to host 10.111.111.14 port 22: No route to host
ssh: connect to host 10.111.111.14 port 22: No route to host
ssh: connect to host 10.111.111.14 port 22: No route to host
ssh: connect to host 10.111.111.14 port 22: No route to host
ssh: connect to host 10.111.111.14 port 22: No route to host
ssh: connect to host 10.111.111.14 port 22: No route to host
ssh: connect to host 10.111.111.14 port 22: No route to host
ssh: connect to host 10.111.111.14 port 22: No route to host
ssh: connect to host 10.111.111.14 port 22: No route to host
ssh: connect to host 10.111.111.14 port 22: No route to host
ssh: connect to host 10.111.111.14 port 22: No route to host
ssh: connect to host 10.111.111.14 port 22: No route to host
ssh: connect to host 10.111.111.14 port 22: No route to host
ssh: connect to host 10.111.111.14 port 22: No route to host
ssh: connect to host 10.111.111.14 port 22: No route to host
ssh: connect to host 10.111.111.14 port 22: No route to host
ssh: connect to host 10.111.111.14 port 22: No route to host
ssh: connect to host 10.111.111.14 port 22: No route to host
ssh: connect to host 10.111.111.14 port 22: No route to host
ssh: connect to host 10.111.111.14 port 22: No route to host
ssh: connect to host 10.111.111.14 port 22: No route to host
ssh: connect to host 10.111.111.14 port 22: No route to host
ssh: connect to host 10.111.111.14 port 22: No route to host
ssh: connect to host 10.111.111.14 port 22: No route to host
ssh: connect to host 10.111.111.14 port 22: No route to host
ssh: connect to host 10.111.111.14 port 22: No route to host
ssh: connect to host 10.111.111.14 port 22: No route to host
ssh: connect to host 10.111.111.14 port 22: No route to host
ssh: connect to host 10.111.111.14 port 22: No route to host
ssh: connect to host 10.111.111.14 port 22: No route to host
ssh: connect to host 10.111.111.14 port 22: No route to host
E> transport closed: internal-error
ssh: connect to host 10.111.111.14 port 22: No route to host
E
======================================================================
ERROR: testBasic (__main__.TestShutdownRestart)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "./check-shutdown-restart", line 39, in testBasic
    m.wait_boot()
  File "/home/hubbot/hubbot/2af96ce5e7a5c9babf3bb0494f71f9e9adec989f_f22_x86-64/test/testvm.py", line 812, in wait_boot
    self.wait_user_login()
  File "/home/hubbot/hubbot/2af96ce5e7a5c9babf3bb0494f71f9e9adec989f_f22_x86-64/test/testvm.py", line 159, in wait_user_login
    raise Failure("Timed out waiting for /run/nologin to disappear")
Failure: Timed out waiting for /run/nologin to disappear

======================================================================
ERROR: testBasic (__main__.TestShutdownRestart)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/hubbot/hubbot/2af96ce5e7a5c9babf3bb0494f71f9e9adec989f_f22_x86-64/test/testlib.py", line 433, in tearDown
    self.copy_journal("FAIL")
  File "/home/hubbot/hubbot/2af96ce5e7a5c9babf3bb0494f71f9e9adec989f_f22_x86-64/test/testlib.py", line 586, in copy_journal
    m.download_dir("/var/log/journal", dir)
  File "/home/hubbot/hubbot/2af96ce5e7a5c9babf3bb0494f71f9e9adec989f_f22_x86-64/test/testvm.py", line 400, in download_dir
    subprocess.check_call(cmd)
  File "/usr/lib64/python2.7/subprocess.py", line 542, in check_call
    raise CalledProcessError(retcode, cmd)
CalledProcessError: Command '['scp', '-i', '/home/hubbot/hubbot/2af96ce5e7a5c9babf3bb0494f71f9e9adec989f_f22_x86-64/test/identity', '-o', 'StrictHostKeyChecking=no', '-o', 'UserKnownHostsFile=/dev/null', '-r', 'root@10.111.111.14:/var/log/journal', 'check-shutdown-restart-testBasic-10.111.111.14-FAIL.journal']' returned non-zero exit status 1

----------------------------------------------------------------------
Ran 1 test in 359.934s
```

http://files.cockpit-project.org/hubbot/2af96ce5e7a5c9babf3bb0494f71f9e9adec989f_f22_x86-64/hubbot.html

**Comment by stefwalter** [comment]
Another case of this:

```
..> couldn't load os data: cancelled
> problem retrieving real time metrics data: authentication-failed
> problem retrieving real time metrics data: authentication-failed
> problem retrieving real time metrics data: authentication-failed
> problem retrieving real time metrics data: authentication-failed
.E
======================================================================
ERROR: testFrameReload (__main__.TestMultiMachine)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "./check-multi-machine", line 75, in setUp
    self.machine2.wait_boot()
  File "/home/hubbot/hubbot/9ccf714b666603d7433dc5522dc289892470457f_f22_x86-64.1/test/testvm.py", line 811, in wait_boot
    raise Failure("Unable to reach machine %s via ssh." % (self.address))
Failure: Unable to reach machine 10.111.111.11 via ssh.

----------------------------------------------------------------------
Ran 4 tests in 483.662s
```

http://files.cockpit-project.org/hubbot/9ccf714b666603d7433dc5522dc289892470457f_f22_x86-64.1/hubbot.html

**Comment by stefwalter** [comment]
I think this is another case of this:

```


OK
Wrote check-shutdown-restart-testBasic-FAIL.png
ssh_exchange_identification: read: Connection reset by peer
Essh_exchange_identification: read: Connection reset by peer
E
======================================================================
ERROR: testBasic (__main__.TestShutdownRestart)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "./check-shutdown-restart", line 40, in testBasic
    self.start_cockpit()
  File "/home/hubbot/hubbot/8015865b0615ee19ae448c54933385d34fdd25bb_r7_x86-64/test/testlib.py", line 481, in start_cockpit
    self.machine.execute("systemctl start cockpit-testing.socket")
  File "/home/hubbot/hubbot/8015865b0615ee19ae448c54933385d34fdd25bb_r7_x86-64/test/testvm.py", line 335, in execute
    raise subprocess.CalledProcessError(proc.returncode, command)
CalledProcessError: Command 'systemctl start cockpit-testing.socket' returned non-zero exit status 255

======================================================================
ERROR: testBasic (__main__.TestShutdownRestart)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/hubbot/hubbot/8015865b0615ee19ae448c54933385d34fdd25bb_r7_x86-64/test/testlib.py", line 433, in tearDown
    self.copy_journal("FAIL")
  File "/home/hubbot/hubbot/8015865b0615ee19ae448c54933385d34fdd25bb_r7_x86-64/test/testlib.py", line 586, in copy_journal
    m.download_dir("/var/log/journal", dir)
  File "/home/hubbot/hubbot/8015865b0615ee19ae448c54933385d34fdd25bb_r7_x86-64/test/testvm.py", line 400, in download_dir
    subprocess.check_call(cmd)
  File "/usr/lib64/python2.7/subprocess.py", line 542, in check_call
    raise CalledProcessError(retcode, cmd)
CalledProcessError: Command '['scp', '-i', '/home/hubbot/hubbot/8015865b0615ee19ae448c54933385d34fdd25bb_r7_x86-64/test/identity', '-o', 'StrictHostKeyChecking=no', '-o', 'UserKnownHostsFile=/dev/null', '-r', 'root@10.111.111.13:/var/log/journal', 'check-shutdown-restart-testBasic-10.111.111.13-FAIL.journal']' returned non-zero exit status 1
```

http://files.cockpit-project.org/hubbot/8015865b0615ee19ae448c54933385d34fdd25bb_r7_x86-64/hubbot.html

**Comment by dperpeet** [comment]
This was fixed in #2759 by using the distinct libvirt reboot event, which doesn't rely on ssh availability.
